File: min-stack/min-stack.cs
public class MinStack {

    private Stack<int> s;
    private Stack<int> min;
    
    public MinStack() {
        s = new Stack<int>();
        min = new Stack<int>();
    }
    
    public void Push(int val) {
        if (s.Count == 0 || val <= min.Peek())
            min.Push(val);
        s.Push(val);
    }
    
    public void Pop() {
        if (s.Peek() == min.Peek())
            min.Pop();
        s.Pop();
    }
    
    public int Top() { return s.Peek(); }
    
    public int GetMin() { return min.Peek(); }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.Push(val);
 * obj.Pop();
 * int param_3 = obj.Top();
 * int param_4 = obj.GetMin();
 */

File: min-stack/min-stack.py
class MinStack:
    
    def __init__(self):
        self.s = []
        self.min = []

    def push(self, val: int) -> None:
        if self.s == [] or val <= self.min[-1]:
            self.min.append(val)
        self.s.append(val)   

    def pop(self) -> None:
        if self.s[-1] == self.min[-1]:
            self.min.pop()
        self.s.pop()

    def top(self) -> int:
        return self.s[-1]

    def getMin(self) -> int:
        return self.min[-1]

# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()
